### PR TITLE
Implemented columnMostCommonValueInSetConstraint

### PR DESCRIPTION
**Pull request by MilenaTrajanoska**
## Description

Implemented columnMostCommonValueInSetConstraint

### General Checklist

* [ ] Tests added for this feature/bug
      if it was a bug, test must cover it.
* [ ] Conform by the style guides, by using formatter
* [ ] Documentation updated
* [ ] (optional) Please add a label to your PR

### CODE REVIEW

File: src/whylogs/core/datasetprofile.py [hunk 1]
```diff
@@ -687,7 +689,15 @@ def apply_summary_constraints(self, summary_constraints: Optional[Mapping[str, S
             if feature_name in self.columns:
                 colprof = self.columns[feature_name]
                 summ = colprof.to_summary()
-                constraints.update(summ.number_summary)
+                frequent_itmes_summ = colprof.frequent_items.to_summary(max_items=1, min_count=1)
```
Comment (Taxuspt) [inline src/whylogs/core/datasetprofile.py]: Maybe typo `frequent_items_summ` ?

File: src/whylogs/core/datasetprofile.py [hunk 2]
```diff
@@ -687,7 +689,15 @@ def apply_summary_constraints(self, summary_constraints: Optional[Mapping[str, S
...
+                most_common_val = frequent_itmes_summ.items[0].json_value if frequent_itmes_summ else None
```
Comment (Taxuspt) [inline src/whylogs/core/datasetprofile.py]: In what condition would this be `None`?

File: src/whylogs/core/statistics/constraints.py [hunk 3]
```diff
@@ -318,23 +352,31 @@ def merge(self, other) -> "SummaryConstraint":
...
+        elif msg.HasField("second_field") and not msg.HasField("value") and not msg.HasField("between") and not msg.HasField("reference_set"):
             return SummaryConstraint(
                 msg.first_field,
                 msg.op,
                 second_field=msg.second_field,
                 name=msg.name,
                 verbose=msg.verbose,
             )
-        elif msg.HasField("between") and not msg.HasField("value") and not msg.HasField("second_field"):
+        elif msg.HasField("between") and not msg.HasField("value") and not msg.HasField("second_field") and not msg.HasField("reference_set"):
```
Comment (Taxuspt) [inline src/whylogs/core/statistics/constraints.py]: This if condition could be turned into a function. Maybe sync with @pecop2 in this

File: tests/unit/core/statistics/test_constraints.py [hunk 4]
```diff
@@ -460,11 +461,72 @@ def test_serialization_deserialization_values_in_set_constraint():
...
+    assert json_value["name"] == f"summary most_common_value {Op.Name(Op.IN)} {val_set}"
+    assert json_value["firstField"] == "most_common_value"
+    assert json_value["op"] == Op.Name(Op.IN)
+    assert json_value["referenceSet"] == list(val_set)
+    assert json_value["verbose"] is True
+
+
+def test_most_common_value_in_set_constraint_wrong_datatype():
+    with pytest.raises(TypeError):
+        columnMostCommonValueInSetConstraint(value_set=2.3, verbose=True)
```
Comment (Taxuspt) [inline tests/unit/core/statistics/test_constraints.py:1305]: Empty line at the end of the document

**Review by Taxuspt — CHANGES_REQUESTED** [review]
Looks good. Just some minor comments

**Review by jamie256 — CHANGES_REQUESTED** [review]
Some conflict to resolve and point this PR to mainline and I'll take another look.

**Review by jamie256 — APPROVED** [review]
LGTM!

**Pull request merged**